Add support for KSZ8091 ethernet PHY (IDFGH-5845)

ma-lalonde opened the pull request:
This is a preamble to adding Energy Efficient Ethernet (EEE) and Wake-on-LAN (WoL) support for the chip.

Also, what would be the best way to introduce these functions? At this time, only the KSZ8091 supports them (I think). However, energy efficiency is becoming a bigger concern and more chips will implements these functions in the future.

The most obvious way that I can see is adding functions `advertise_eee_ability(bool enable)` and `enable_wol(bool enable, int phy_pin)` to structure `esp_eth_phy_t` (and other parameters if I realise they are necessary).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/espressif/esp-idf?pullRequest=7547) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/espressif/esp-idf?pullRequest=7547) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/esp-idf?pullRequest=7547) it.</sub>

Alvin1Zhang commented:
Thanks for your contribution.

ma-lalonde commented:
I simplified the implementation since the initialization of the KSZ8091 is identical to that of the KSZ8081. Further simplification to KSZ80xx is not (easily) possible due to the registers of the KSZ8041 being differently arranged.

As for the implementation, due to WoL and EEE still being very hardware specific (and EEE still being full of bugs), I will leave those out of the PHY driver and to be implemented by the application programmers.

Note that the KSZ8091 implements MMD registers that the KSZ8081 doesn't, so they can be differentiated at this level. Writing to MMD registers for WoL and EEE configuration implies a specific sequence of PHY register read/writes detailed in the datasheet. I think this sequence is too specific to be implemented at the driver level either and should be left to application developer.

TL;DR: Simplified KSZ8091 implementation and WoL and EEE will be left out to application developer.

kostaond commented on components/esp_eth/include/esp_eth_phy.h:
```diff
@@ -304,6 +297,20 @@ esp_eth_phy_t *esp_eth_phy_new_ksz8041(const eth_phy_config_t *config);
+*/
+static inline esp_eth_phy_t *esp_eth_phy_new_ksz8091(const eth_phy_config_t *config)
+{
+	return esp_eth_phy_new_ksz8081(config);
```
It was found the `tab` is used instead of spaces during the internal review. Could you please fix it?

ma-lalonde replied:
Done.
